servant-client-core/src/Servant/Client/Core/Request.hs

```diff
@@ -135,6 +138,14 @@ addHeader :: ToHttpApiData a => HeaderName -> a -> Request -> Request
 addHeader name val req
   = req { requestHeaders = requestHeaders req Seq.|> (name, toHeader val)}
 
+concatQueryString :: ToForm a
+                  => a
+                  -> Request
+                  -> Request
+concatQueryString form req
+  = let querySeq = Seq.fromList . parseQuery . LBS.toStrict . urlEncodeAsForm $ form
```

Comment:
this encoding than parsing (and later re-encoding) is phishy. Should there be some function in `http-api-data` to avoid that?

Reply:
This is a fair point. See new version. There's a function, as follows:

```toListStable :: Form -> [(Text, Text)] ```

So we can take the output and turn it into `[QueryItem]` and add those to the Sequence.